pkg/cloud/cloudpb/external_storage.go

```diff
@@ -59,3 +64,35 @@ func (m *ExternalStorage) AccessIsWithExplicitAuth() bool {
 		return false
 	}
 }
+
+const assumeRoleProviderExternalIDParam = "external_id"
+
+// EncodeAsString returns the string representation of the provider to be used
+// in URIs.
+func (p ExternalStorage_AssumeRoleProvider) EncodeAsString() string {
+	if p.Role == "" {
+		return ""
+	}
+
+	if p.ExternalID == "" {
+		return p.Role
+	}
+
+	return fmt.Sprintf("%s;%s=%s", p.Role, assumeRoleProviderExternalIDParam, p.ExternalID)
+
+}
+
+// DecodeRoleProviderString decodes a string into an
+// ExternalStorage_AssumeRoleProvider.
+func DecodeRoleProviderString(roleProviderString string) (p ExternalStorage_AssumeRoleProvider) {
```

Comment:
I don't want to relitigate if it has already been discussed but have we arrived at assume_role;external_id=123 as the most appropriate way we expect users to pass in this ID? Why not something simpler like assume_role:123?

Reply:
The discussion can be found here:https://cockroachlabs.slack.com/archives/C02Q8MLFFE0/p1665778499688029

I don't think there was much back and forth about this as the current implementation of separating with  `;` seemed good to me. We can't use a `:` in the string as ARNs also contain `:`. We may also want to add other fields in the future, with duration of short-lived credentials being probably the top one.

Reply:
makes sense, thanks!